src/javaRosa.js

```diff
@@ -1055,6 +1054,85 @@ define([
 
     };
 
+    var itextMarkdownWidget = function (mug, language, form, options) {
+        options = options || {};
+        var parent = options.parent;
+        var widget = itextLabelWidget(mug, language, form, options),
+            super_setValue = widget.setValue,
+            super_getUIElement = widget.getUIElement,
+            super_handleChange = widget.handleChange,
+            wantsMarkdown = true,
+            markdownOff, markdownOn, markdownOutput;
+
+        widget.toggleMarkdown = function() {
+            parent.toggleClass("has-markdown");
+        };
+
+        markdownOutput = $('<div>').addClass("controls well markdown-output");
+
+        widget.handleChange = function() {
+            super_handleChange();
+            var val = widget.getValue(),
+                item = this.getItextItem();
+            if (/[~*#[\]]+|^-/m.test(val)) {
```

Comment:
maybe extract this test to a function (i think the intent is `looksLikeMarkdown(val)`?) so it's clear what the intent is.